content/number-theory/Eratosthenes.h

```diff
@@ -0,0 +1,21 @@
+/**
+ * Author: Brian Grana
+ * Date: 2024-10-02
+ * Source: own
+ * Description: Prime sieve that generates a boolean vector where
+ * sieve[i] = 1 if and only if the number is prime.
+ * Time: $O(N\log\log (N))$
+ * Status: Tested on https://codeforces.com/gym/102953/problem/8
+ */
+#pragma once
+
+vector<bool> sieve(int n) {
+	vector<bool> sv(n+1, 1);
+	sv[0] = 0, sv[1] = 0;
+	for (int i = 2; i * i <= n; i++) {
+		if (sv[i]) {
+			for (int j = i * i; j <= n; j += i) sv[j] = 0;
+		}
+	}
```

Comment:
do we want to compact this a little bit? i.e.
```cpp
for (int i = 2; i * i <= n; i++) if (sv[i]) {
    for (int j = i * i; j <= n; j += i) sv[j] = 0;
}
```

Reply:
fixed